web-marketplace/src/components/organisms/EditProfileForm/EditProfileForm.tsx

```diff
@@ -112,6 +99,16 @@ const EditProfileForm: React.FC<React.PropsWithChildren<EditProfileFormProps>> =
       }
     }, [isDirtyRef, isDirty]);
 
+    const isFormUpdated: boolean = Object.keys(formValues).some(
+      key =>
+        formValues[key as FormValuesKeys] !==
+        initialValues[key as FormValuesKeys],
+    );
```

Comment:
I don't think we need this custom implementation, we already have `isDirtyRef` which comes from the parent component

Reply:
`isDirtyRef` is not returning the right state when the form is updated and then is reverted to it's original state. Maybe I can update `isDirtyRef` implementation, let me have a look.

Reply:
what do you mean by "not the right state"?

Reply:
`isDirtyRef` incorrectly marks the form as dirty even after changes are undone. For example, if the profile name is changed and then reverted to the original value without saving, the form remains dirty (`isDirtyRef.current: true`)

Reply:
I think the issue is that it's a ref because if you inspect `isDirty` from `useFormState({
      control: form.control,
    });` in `EditProfileForm`, it works almost as expected (ie changing back to false if we come back to the initial values), except for the profile type for some reason... I guess we could use `dirtyFields` instead from `useFormState`, this one seems to work fine.